core/src/main/java/org/testcontainers/containers/ContainerState.java

```diff
@@ -35,6 +37,17 @@ default Boolean isRunning() {
         }
     }
 
+    /**
+     * @return has the container health state 'healthy'?
+     */
+    default Boolean isHealthy() {
+        try {
+            return getContainerId() != null && DockerClientFactory.instance().client().inspectContainerCmd(getContainerId()).exec().getState().getHealth().getStatus().equals(STATE_HEALTHY);
```

Comment:
could you please extract the result of `DockerClientFactory.instance().client().inspectContainerCmd(getContainerId()).exec()`?

Reply:
Gladly, I simply copied this from the `isRunning()` method and thought this code was leveraging lazy evaluation of `&&` (cargo cult 🛩).

Reply:
yeah, but that was an old code and we should improve bit by bit :)

Reply:
I tried to do a little refactoring here and wanted to simply use `getContainerInfo()`, but this seems to be implemented by the getter of the `GenericContainer`, which is only initially set during start in the `tryStart()` method. So my question, is it intended for this public API to basically return a cached version of `InspectContainerResponse` or am I missing something?

Reply:
Thanks for the clarification. I've thus added the `getCurrentContainerInfo()` default method to the interface. I would have been fine with a private method for this, since I was mostly for avoiding code duplication, but I think it's actually useful to also have this as a public API?